### PR TITLE
Remove Testcase ontology

### PR DESCRIPTION
**Pull request by gaurav**
We use some RDF properties in the `testcase:` namespace to tie the JSON-LD file together (since keys in JSON-LD files without an associated RDF property will not be read as RDF). However, these properties don't actually need to be defined anywhere, certainly not in the Phyloref Testcase Ontology that we created ages ago as a repository for these terms. This PR removes the Phyloref Testcase Ontology from the OWL imports in generated phyloreferences.

It also removes many of the uses of the `testcase:` namespace, including the no-longer-used taxonomic unit matching system. Three of these terms could be replaced with CDAO terms, so I've replaced these in this PR: 
- `testcase:PhyloreferenceTestPhylogeny` corresponds to a [`CDAO:RootedTree`](http://purl.obolibrary.org/obo/CDAO_0000012).
- `testcase:has_root_node` corresponds to [`CDAO:has_Root`](http://purl.obolibrary.org/obo/CDAO_0000148).
- `testcase:has_node` corresponds to [`CDAO:has_Element`](http://purl.obolibrary.org/obo/CDAO_0000198) (confusingly, it does NOT correspond to [`CDAO:has_Node`](http://purl.obolibrary.org/obo/CDAO_0000162), whose domain is Edges, not Trees).

As of https://github.com/phyloref/phyx.js/issues/78#issuecomment-792373978, there are five `testcase:` terms remaining in Phyx.js:
* Needed to tie the generated OWL/JSON-LD file together:
  * `testcase:has_phylogeny`, in the form `<ontology> testcase:has_phylogeny <phylogeny>`.
  * `testcase:has_phyloreference`, in the form `<ontology> testcase:has_phyloreference <phyloref>`.
  * `testcase:has_component_class`, in the form `<phyloref> testcase:has_component_class <component class>`.
* Optional; only needed if we want to keep the verbatim specifiers in the generated OWL/JSON-LD file:
  * `testcase:internal_specifier`, in the form `<phyloref> testcase:internal_specifier <internal specifier>`.
  * `testcase:external_specifier`, in the form `<phyloref> testcase:external_specifier <external specifier>`.

We could figure out what to do with these terms in this PR, or leave that for a separate PR.

### CODE REVIEW

File: src/wrappers/PhylogenyWrapper.js [hunk 1]
```diff
@@ -431,7 +431,7 @@ class PhylogenyWrapper {
     // Set name and class for phylogeny. If no '@id' is set, use the provided
     // fallbackIRI.
     if (!has(phylogenyAsJSONLD, '@id')) phylogenyAsJSONLD['@id'] = fallbackIRI;
-    phylogenyAsJSONLD['@type'] = owlterms.PHYLOREFERENCE_PHYLOGENY;
+    phylogenyAsJSONLD['@type'] = owlterms.CDAO_ROOTED_TREE;
```
Comment (hlapp) [inline src/wrappers/PhylogenyWrapper.js:434]: How about http://ontology.phyloref.org/phyloref.owl#ReferencePhylogenyEvidence. Wouldn't that be more appropriate? It's not currently asserted as a subclass of CDAO:rooted_tree, but that could be added.

**Review by hlapp — COMMENTED** [review]
See inline comment. I'm also wondering whether CDAO doesn't simply need to be fixed so as not to assert an edge as the domain of `has_Node`.

File: src/wrappers/PhylogenyWrapper.js (hunk 1 above)
Comment (gaurav) [reply src/wrappers/PhylogenyWrapper.js]: Yes, that's better! I've made that change in 640c4cb4bf09.

**Comment by gaurav** [comment]
> I'm also wondering whether CDAO doesn't simply need to be fixed so as not to assert an edge as the domain of `has_Node`.

I don't *think* so, since `has_Node` (and its subproperties, `has_Parent_Node` and `has_Child_Node`) are designed specifically for use with edges, rather than the node-based `has_Parent`/`has_Child`. But I've opened an issue to discuss this on CDAO: https://github.com/evoinfo/cdao/issues/17.

**Review by hlapp — APPROVED** [review]
👍🏻

**Pull request merged**